src/extension.ts

```diff
@@ -1618,17 +1618,15 @@ export async function activate(
 ): Promise<API> {
     const api = new API();
 
-    // deferred activation in case activationCfg is set by peer extension synchronously
-    setTimeout(async () => {
-        await zephyr.activate(api.activationCfg.topdir);
-        await typeLoader.activate(context);
-
-        const engine = new DTSEngine();
-        await engine.loadCtxs();
-        await dts.parser.activate(context);
-        engine.activate(context);
-        treeView.activate(context);
-    }, 1);
+    await zephyr.activate();
```

Comment:
This will present an error if Zephyr isn't found, which was the motivation for the original timeout mechanism, I believe. I think @bencefr added this after I went on vacation, and probably for a good reason. Is it safe to do it this way?

Reply:
The timeout has been added so the device-tree extension can be used both standalone and with main extension. The idea is that activating all the internals is not crucial to be done for vscode, and if it is performed right after _activate()_ has returned to the main extension, that will have a chance to configure/setup this extension synchronously before this starts to complain about settings.

Reply:
My point is that the modification here shall not restrict this extension to be used independently, but should make the experience smooth while using it in combination of the main extension.